_posts/2020-09-30-ghelamco-alert.md

```diff
@@ -0,0 +1,824 @@
+This means that after 15 minutes of not receiving this metric, the cloudwatch alarm will trigger and notifies us via email that the RPI is either down or disconnected from the internet.  
+This enables us to respond quickly and take action to restore connectivity to the device.  
+
+<div style="text-align: center;">
+  <img alt="Ghelamco-alert Cloudwatch Alarm" src="/img/2020-09-25-ghelamco-alert/ghelamco-rpi-backend-down-alert.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+##### Cloudwatch logs
+{:.no_toc}
+
+The logs from our spring boot application get sent to Cloudwatch logs. 
+This allows us to check the log files in the AWS cloud without needing access to the RPI device itself.  
+<div style="text-align: center;">
+  <img alt="Ghelamco-alert Cloudwatch Alarm" src="/img/2020-09-25-ghelamco-alert/cloudwatch-logs.PNG" width="auto" height="auto" target="_blank" class="image fit">
+</div>
+
+
+#### AWS IoT
+{:.no_toc}
+
+The reason for using AWS IoT is for the sake of "being connected to the cloud".  
+When we are connected to the cloud, we are able to communicate with our device "through the cloud", meaning from anywhere we want!   
+In our case this means that we want our RPI device to be connected to AWS IoT whenever it is up and running and has internet connectivity.  
+Of course the communication between our RPI and the AWS cloud has to be secured.
+AWS IoT uses authentication and authorization workflow by using certificates that you issue from AWS IoT and upload to your edge device.  
+We treat our RPI as an edge device in this project.
+
+Authentication works based on the certificates that the edge device presents to AWS IoT.  
+Once the certificate authentication is successful, AWS IoT checks which policies got attached to those certificates to grant it specific authorizations within the AWS cloud.  
+
+##### Authentication: certificates
+{:.no_toc}
+
+When you add a new edge device in AWS IOT it is called "a thing".  
+When registering "a thing" it generates 2 keys for us: a public key and a private key.  
+AWS IoT will provide a certificate signing request for the public key, which will sign the generated certificate with the root certificate's private key.  
+
+Our thing-certificate and our private key are our credentials when we try to communicate with AWS IoT to access our edge device.  
+The only additional input that we need to provide on top of the generated signing request is the root certificate to check the signing.  
+
+* Certificates
+* private Key
+* Root CA
+
+Since these certificates get generated per thing that you register they should only be used for 1 device.  
+When I developed the application I had to register another thing for my laptop.  
+Connecting from the RPI and the laptop with the same certificates caused some unwanted behavior like connection interrupts.  
```

Comment:
Can this be detected in aws iot?

Would you be able to take over a connection by having another device send in data?

Just some food for thought

Reply:
yeah, if you posses the certificates you have the keys to the connection. best practice is probably not to share certificates but it was easier this way in our dev phase